Clean up the React app

wpalani opened the pull request:
## Proposed changes
Since we moved a little bit quickly in 3.0 release, we didn't have enough time to clean up old code and structure the app properly, also we received new components/code from other teams with little time to give feedback. This PR aims to clean up the code of the React app including:

- Unified function definitions (Arrow functions).
- Remove old components since they're not being used anymore.
- Organize module imports throughout the app's components and files.
- Introduce ESLint and lint the app using `plugin:@wordpress/eslint-plugin/recommended`.
- Add ESLint Github action for future pushes and PRs.

Todo:

- Add a proper preloader to the app. Right now, the app loads in a blank screen. most likely, we need something PHP side since the app grew larger recently.
- Clean up the error catcher/boundaries and the error template.

<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->

## Type of Change

<!-- What types of changes does your code introduce? -->
<!-- _Put an `x` in the boxes that apply_ -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

<!-- _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._ -->

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

circlecube commented on src/app/data/routes.js:
```diff
@@ -18,15 +17,15 @@ import Settings from '../pages/settings';
 import Staging from '../pages/staging';
 import Help from '../pages/help';
 import { getMarketplaceSubnavRoutes } from '../../../vendor/newfold-labs/wp-module-marketplace/components/marketplaceSubnav';
```
For these types of imports, looks like we'll need to add `composer install` to the workflow so that the modules are installed and can be found.

bradp replied:
Should we configure an alias in a `jsconfig.js` file? Would let us a an `import { foo } from @composer/bar/baz` rather than this triple `../`.

circlecube replied:
I like this idea, and I'll create a new PR for it.

circlecube commented on src/app/pages/example/index.js:
```diff
@@ -1,44 +1,86 @@
-import { SectionContainer, SectionHeader, SectionContent } from "../../components/section";
-import { RocketLaunchIcon } from "@heroicons/react/24/outline";
-import { Page } from "../../components/page";
+import { Button } from '@newfold/ui-component-library';
```
Should probably delete this file/dir too since it isn't used.

bradp commented on src/app/pages/ecommerce/page.js:
```diff
@@ -1,45 +1,51 @@
+			bluehostSettingsApiFetch(
+				{ comingSoon: ! store.comingSoon },
+				// eslint-disable-next-line no-console
+				console.error,
```
🔢 Similar to my comment about the `console.warn`, is this helpful to have or just a hold-over from dev?

bradp commented on src/app/pages/help/index.js:
```diff
@@ -1,59 +1,61 @@
-					'wp-plugin-bluehost')}
+					title={ __( 'Help', 'wp-plugin-bluehost' ) }
+					subTitle={ __(
+						'We are available 24/7 to help answer questions and solve your problems.',
```
We use "24 hours - 7 days" elsewhere, should these be consistent?

_If this text comes from marketing or somewhere else, feel free to ignore._

bradp commented on src/app/pages/home/helpCard.js:
```diff
@@ -1,26 +1,34 @@
+					/>
+					<p>From DIY to full-service help.</p>
+					<p>
+						Call or chat 24/7 for support or let our experts build
```
🔢 "24/7" vs "24 hours - 7 days"

bradp commented on src/app/pages/home/helpCard.js:
```diff
@@ -1,26 +1,34 @@
+						href="#/help"
+						className="wppbh-help-link"
+					>
+						Help me
```
Does it make more sense to just have this be "Help" or "Get Help"? 
"Help me" somewhat makes me think it's the system asking for help.

> 🤖 _help me i am trapped in this computer_

_If this text comes from marketing or somewhere else, feel free to ignore._

wpalani replied:
Addressed in this commit: https://github.com/bluehost/bluehost-wordpress-plugin/pull/820/commits/f2189f1882d8bc9b38edf951a32c2bcc2d63133f

bradp commented on src/app/pages/home/welcomeSection.js:
```diff
@@ -1,23 +1,28 @@
+		bluehostSettingsApiFetch(
+			{ comingSoon: ! store.comingSoon },
+			// eslint-disable-next-line no-console
+			console.error,
```
🔢 necessary `console.error`?

bradp commented on src/app/pages/marketplace/index.js:
```diff
@@ -2,54 +2,64 @@ import apiFetch from '@wordpress/api-fetch';
+				'bluehost-wordpress-plugin'
+			),
+			error: __(
+				'Oops, there was an error loading the marketplace, please try again later.',
+				'bluehost-wordpress-plugin'
+			),
+			noProducts: __(
+				'Sorry, no marketplace items. Please, try again later.',
+				'bluehost-wordpress-plugin'
```
Can we make the "Please try again later" consistent?

I'd suggest `, please try again later.` for both.

bradp commented on src/app/pages/marketplace/index.js:
```diff
@@ -2,54 +2,64 @@ import apiFetch from '@wordpress/api-fetch';
+				'bluehost-wordpress-plugin'
+			),
+			noProducts: __(
+				'Sorry, no marketplace items. Please, try again later.',
```
Can we change this to "Sorry, no marketplace items **found**" to be more clear?

bradp commented on src/app/pages/settings/comingSoon.js:
```diff
@@ -1,115 +1,131 @@
-				'Coming soon page is active. Site requires login.',
-				'wp-plugin-bluehost'
-			)
+					'Coming soon page is active. Site requires login.',
```
```suggestion
					'Coming soon page is active. Site is only accessible to logged in users.',
```

bradp commented on src/app/pages/settings/comingSoon.js:
```diff
@@ -1,115 +1,131 @@
-				'Coming soon page is not active. Site is live to visitors.',
-				'wp-plugin-bluehost'
-			);
+					'Coming soon page is not active. Site is live to visitors.',
```
```suggestion
					'Coming soon page is not active. Site is public and accessible to all visitors.',
```

bradp commented on src/app/pages/settings/comingSoon.js:
```diff
@@ -1,115 +1,131 @@
 					<Alert variant="info">
-						{__('Your website is currently displaying a "Coming Soon" page.', 'wp-plugin-bluehost')}
+						{ __(
+							'Your website is currently displaying a "Coming Soon" page.',
```
```suggestion
							'Your site is currently displaying a "Coming Soon" page.',
```

If we prefer "website", then there's a few other places we should update from "site" → "website" for consistency.

bradp commented on src/app/pages/settings/comingSoon.js:
```diff
@@ -1,115 +1,131 @@
 					<Alert variant="error">
-						{__('Oops! Something went wrong. Please try again.', 'wp-plugin-bluehost')}
+						{ __(
+							'Oops! Something went wrong. Please try again.',
```
🔢 Related to previous comments about "please try again", let's have this match whatever gets decided there.

bradp commented on src/app/pages/settings/commentSettings.js:
```diff
@@ -1,137 +1,150 @@
-			__('Comments on posts are disabled after ', 'wp-plugin-bluehost') +
+			__(
+				'Comments on posts are disabled after ',
+				'wp-plugin-bluehost'
+			) +
 			closeCommentsDays +
-			_n(' day.', ' days.', parseInt(closeCommentsDays), 'wp-plugin-bluehost')
+			_n(
+				' day.',
+				' days.',
+				parseInt( closeCommentsDays ),
+				'wp-plugin-bluehost'
```
This should be using `sprintf` to fill the `closeCommentsDays` and the `day`/`days`, not all languages will follow the same grammatical format as English.

bradp commented on src/app/pages/settings/commentSettings.js:
```diff
@@ -143,105 +156,115 @@ const CloseCommentsDays = ({ setError, notify }) => {
 		//`Posts will display ${commentsPerPage} comments at a time.`
 		return (
-			__('Posts will display ', 'wp-plugin-bluehost') +
+			__( 'Posts will display ', 'wp-plugin-bluehost' ) +
```
🔢 translation strings

bradp commented on src/app/pages/settings/commentSettings.js:
```diff
@@ -143,105 +156,115 @@ const CloseCommentsDays = ({ setError, notify }) => {
-			selectedLabel={commentsPerPage}
-			options={[
+			label={
+				__( 'Display ', 'wp-plugin-bluehost' ) +
```
🔢 translation strings

bradp commented on src/app/pages/settings/contentSettings.js:
```diff
@@ -1,181 +1,217 @@
-			__('Saving drafts and updating published content creates revisions. Make changes with confidence, knowing you can take ', 'wp-plugin-bluehost') +
+			__(
+				'Saving drafts and updating published content creates revisions. Make changes with confidence, knowing you can take ',
+				'wp-plugin-bluehost'
+			) +
 			contentRevisions +
-			_n(' step back.', ' steps back.', parseInt(contentRevisions), 'wp-plugin-bluehost')
+			_n(
+				' step back.',
+				' steps back.',
```
🔢 translation strings

bradp commented on src/app/pages/settings/contentSettings.js:
```diff
@@ -1,181 +1,217 @@
-			selectedLabel={contentRevisions}
-			options={[
+			label={ __(
+				'Number of revisions posts can save ',
```
```suggestion
				'Number of post revisions to save ',
```

bradp replied:
```suggestion
				'Number of saved post revisions  ',
```

bradp commented on src/app/pages/settings/contentSettings.js:
```diff
@@ -1,181 +1,217 @@
-			__('The trash will automatically empty every ', 'wp-plugin-bluehost') +
+			__(
+				'The trash will automatically empty every ',
+				'wp-plugin-bluehost'
+			) +
 			numTrashWeeks +
-			_n( ' week.', ' weeks.', parseInt(numTrashWeeks), 'wp-plugin-bluehost' )
+			_n(
+				' week.',
+				' weeks.',
+				parseInt( numTrashWeeks ),
+				'wp-plugin-bluehost'
+			)
```
🔢 translation strings

bradp commented on src/app/pages/settings/contentSettings.js:
```diff
@@ -1,181 +1,217 @@
-			numTrashWeeks +
-			 _n( ' week.', ' weeks.', parseInt(numTrashWeeks), 'wp-plugin-bluehost' )
+				__(
+					'The trash will automatically empty every ',
+					'wp-plugin-bluehost'
+				) +
+				numTrashWeeks +
+				_n(
+					' week.',
+					' weeks.',
+					parseInt( numTrashWeeks ),
+					'wp-plugin-bluehost'
+				)
```
🔢 translation strings

bradp commented on src/app/pages/settings/contentSettings.js:
```diff
@@ -1,181 +1,217 @@
-			description={__('Controls for content revisions and how often to empty the trash.', 'wp-plugin-bluehost')}
+			title={ __( 'Content Options', 'wp-plugin-bluehost' ) }
+			description={ __(
+				'Controls for content revisions and how often to empty the trash.',
```
We say "content revisions" here but "post revisions" elsewhere, let's make it all consistent. 

I prefer "content", but I think that implies that it's _all_ content, including things like pages or site tagline or menus or something. 🤔 Not sure what's best.

bradp commented on src/app/pages/settings/index.js:
```diff
@@ -2,36 +2,52 @@ import AutomaticUpdates from './automaticUpdates';
+				<SectionHeader
+					title={ __( 'Settings', 'wp-plugin-bluehost' ) }
+					subTitle={ __(
+						'This is where you can manage common settings for your website.',
```
🔢 "this is where you can manage X" vs "Manage X"

bradp commented on src/app/components/site-info/index.js:
```diff
@@ -1,62 +1,78 @@
+					>
+						<WordPressIcon />
+						{ isWooCommerce && isStore
+							? 'View Store'
```
Need to make this translatable

wpalani replied:
Addressed in this commit: https://github.com/bluehost/bluehost-wordpress-plugin/pull/820/commits/f2189f1882d8bc9b38edf951a32c2bcc2d63133f

bradp commented on src/app/components/site-info/index.js:
```diff
@@ -1,62 +1,78 @@
+						className="nfd-bg-[#383F4A] nfd-text-tiny nfd-w-full min-[400px]:nfd-w-auto"
+					>
+						<BluehostIcon />
+						Bluehost Account
```
:1234: Need to make this translatable

wpalani replied:
Addressed in this commit: https://github.com/bluehost/bluehost-wordpress-plugin/pull/820/commits/f2189f1882d8bc9b38edf951a32c2bcc2d63133f

bradp commented on src/app/components/webinars-banner/index.js:
```diff
@@ -1,87 +1,107 @@
-			nextWebinar.link = nextWebinar.link ?? 'https://www.bluehost.com/blog/events/';
+			nextWebinar.link =
+				nextWebinar.link ?? 'https://www.bluehost.com/blog/events/';
 			nextWebinar.ctaText = nextWebinar.ctaText ?? 'Register Now';
```
:1234: Need to make this translatable

bradp commented on src/app/pages/home/helpCard.js:
```diff
@@ -1,26 +1,34 @@
+						src={ SupportIllustration }
+						alt="Help Agent Illustration"
+					/>
+					<p>From DIY to full-service help.</p>
```
:1234: Need to make this translatable

wpalani replied:
Addressed in this commit: https://github.com/bluehost/bluehost-wordpress-plugin/pull/820/commits/f2189f1882d8bc9b38edf951a32c2bcc2d63133f

bradp commented on src/app/pages/settings/commentSettings.js:
```diff
@@ -143,105 +156,115 @@ const CloseCommentsDays = ({ setError, notify }) => {
 					<Alert variant="error">
-						{__('Oops! Something went wrong. Please try again.', 'wp-plugin-bluehost')}
+						{ __(
+							'Oops! Something went wrong. Please try again.',
```
🔢 please try again consistency

bradp commented on src/app/util/helpers.js:
```diff
@@ -4,63 +4,11 @@ import apiFetch from '@wordpress/api-fetch';
  *
- * @param  string text text for notice
- * @param  text
+ * @param {string} text text for notice
```
```suggestion
 * @param {string} text Text for notice.
```

bradp commented on src/app/util/helpers.js:
```diff
@@ -81,10 +29,9 @@ export const dispatchUpdateSnackbar = ( text = 'Settings Saved' ) => {
- * @param passError setter for the error in component
- * @param thenCallback method to call in promise then
- * @return apiFetch promise
+ * @param {Object}   data         object of data
+ * @param {Function} passError    setter for the error in component
+ * @param {Function} thenCallback method to call in promise then
```
```suggestion
 * @param {Object}   data         Object of data.
 * @param {Function} passError    Setter for the error in component.
 * @param {Function} thenCallback Method to call in promise then.
```

bradp commented on src/app/util/helpers.js:
```diff
@@ -104,10 +51,9 @@ export const bluehostSettingsApiFetch = ( data, passError, thenCallback ) => {
- * @param passError setter for the error in component
- * @param thenCallback method to call in promise then
- * @return apiFetch promise
+ * @param {Object}   data         object of data
+ * @param {Function} passError    setter for the error in component
+ * @param {Function} thenCallback method to call in promise then
```
```suggestion
 * @param {Object}   data         Object of data.
 * @param {Function} passError    Setter for the error in component.
 * @param {Function} thenCallback Method to call in promise then.
```

bradp commented on src/app/util/helpers.js:
```diff
@@ -134,11 +81,11 @@ export const comingSoonAdminbarToggle = ( comingSoon ) => {
-		comingsoonadminbar.style.color = "#E01C1C";
-		comingsoonadminbar.textContent = "Coming Soon";
+		comingsoonadminbar.style.color = '#E01C1C';
+		comingsoonadminbar.textContent = 'Coming Soon';
```
🔢 translation strings

wpalani commented:
@bradp Thanks for your feedback. I addressed some of your comments in this commit: https://github.com/bluehost/bluehost-wordpress-plugin/pull/820/commits/f2189f1882d8bc9b38edf951a32c2bcc2d63133f

I think most of what's left is a language/copy change that we should address in a different PR in a more thought-out way. Since this language is being used across other brand plugins, so we don't have to go back and change it again.
